src/test/java/io/github/enkarin/bookcrossing/init/GreenMailInitializer.java

```diff
@@ -0,0 +1,31 @@
+package io.github.enkarin.bookcrossing.init;
+
+import org.springframework.boot.test.util.TestPropertyValues;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.utility.DockerImageName;
+
+public class GreenMailInitializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
+
+    private static final DockerImageName IMAGE = DockerImageName.parse("greenmail/standalone:1.6.9");
+    public static final GenericContainer<?> GREEN_MAIL_CONTAINER = new GenericContainer<>(IMAGE);
+
+    static {
+        GREEN_MAIL_CONTAINER
+                .waitingFor(Wait.forLogMessage(".*Starting GreenMail standalone.*", 1))
+                .withEnv("GREENMAIL_OPTS", "-Dgreenmail.startup.timeout=5000 -Dgreenmail.setup.test.smtp " +
+                        "-Dgreenmail.hostname=0.0.0.0 -Dgreenmail.users=duke:springboot")
+                .withExposedPorts(3025)
```

Comment:
Try to remove hardcoded port `.withExposedPorts(3025)`

Reply:
java.lang.IllegalStateException: Container doesn't expose any ports